### ISSUE TITLE
Add support for EndpointAccessMode in AWS::ApiGateway::DomainName & AWS::ApiGateway::DomainNameV2

### ISSUE THREAD

**Issue by TaherKapasi** [aws-cloudformation/cfn-lint#4306]
### CloudFormation Lint Version

cfn-lint 1.40.4

### What operating system are you using?

Mac

### Describe the bug

When running `cfn-lint` on a SAM template with an `AWS::ApiGateway::DomainName` or `AWS::ApiGateway::DomainNameV2` resource configured with a [Security Policy](https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-resource-apigateway-domainname.html#cfn-apigateway-domainname-endpointaccessmode) set to `SecurityPolicy_TLS13_1_2_FIPS_PQ_2025_09` one has to add `EndpointAccessMode: STRICT` in order for CloudFormation to successfully deploy.

The problem is `cfn-lint` says the template has an `E3002: Resource properties are invalid`.

Short term workaround was to add E3002 to `ignore_checks`.

### Expected behavior

No errors reported

### Reproduction template

Create a template that include an API

```
  ApiGateway:
    Type: AWS::Serverless::Api
    Properties:
      ...

  ApiDomain:
    Type: AWS::ApiGateway::DomainName
    Properties:
      DomainName: x.y.com
      RegionalCertificateArn: !GetAtt Cert.Arn
      EndpointConfiguration:
        Types:
          - REGIONAL
      SecurityPolicy: SecurityPolicy_TLS13_1_2_FIPS_PQ_2025_09
      EndpointAccessMode: STRICT
```
